src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DefaultOutputFormatterSelector.cs

```diff
@@ -111,24 +117,31 @@ public override IOutputFormatter SelectFormatter(OutputFormatterCanWriteContext
                         contentTypes);
                 }
 
-                if (selectedFormatter == null && !_returnHttpNotAcceptable)
+                if (selectedFormatter == null)
                 {
                     _logger.NoFormatterFromNegotiation(acceptableMediaTypes);
 
-                    selectFormatterWithoutRegardingAcceptHeader = true;
+                    if (!_returnHttpNotAcceptable)
+                    {
+                        selectFormatterWithoutRegardingAcceptHeader = true;
+                    }
                 }
             }
 
             if (selectFormatterWithoutRegardingAcceptHeader)
             {
                 if (contentTypes.Count == 0)
                 {
+                    _logger.SelectingOutputFormatterWithoutUsingContentTypes();
```

Comment:
It looks to me like this one is also redundant. `SelectFormatterNotUsingContentType` also logs and provides more detail.